xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProvider.java

```diff
@@ -129,6 +449,67 @@ abstract MeshCaCertificateProvider create(
         BackoffPolicy.Provider backoffPolicyProvider,
         long renewalGracePeriodSeconds,
         int maxRetryAttempts,
-        GoogleCredentials oauth2Creds);
+        GoogleCredentials oauth2Creds,
+        ScheduledExecutorService scheduledExecutorService,
+        TimeProvider timeProvider,
+        long rpcTimeoutMillis);
   }
+
+  private class ZoneInfoClientInterceptor implements ClientInterceptor {
+    private final String zone;
+
+    ZoneInfoClientInterceptor(String zone) {
+      this.zone = zone;
+    }
+
+    @Override
+    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+        MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+      return new ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(
+          next.newCall(method, callOptions)) {
+
+        @Override
+        public void start(Listener<RespT> responseListener, Metadata headers) {
+          headers.put(KEY_FOR_ZONE_INFO, zone);
+          super.start(
+              new ForwardingClientCallListener.SimpleForwardingClientCallListener<RespT>(
```

Comment:
If you don't override any method of the listener, no need to wrap.

Reply:
Done